doc/1.4/language.md

```diff
@@ -609,10 +609,15 @@ attribute `a`, then one can access it as follows:
 log info: "the value of attribute a is %d", dev.a.val;
 ```
 
-The [`init` template](dml-builtins.html#init) is often useful together with
-`attribute` objects &mdash; in particular in combination with these standard
-templates &mdash; in order to initialize the state associated with the
-`attribute` object.
+These templates also provide an `init_val` parameter and an overridable
+implementation of [`init`](dml-builtins.html#init) that leverages said
+parameter to initialize the `val` session variable (with the default
+being zero-initalizising the variable.)
+
+The `init` template and associated method is often useful together with
+`attribute` objects to initialize any associated state &mdash; even with the
+standard templates mentioned above, as `init_val` cannot be used if the desired
+initial value isn't constant.
```

Comment:
Need to be weakened, it's no longer "often useful". Maybe "init_val can only be used to set a constant initial value; for a non-constant init value you must override `init`"

Reply:
Right, I missed that you're talking about attrs in general. How about reordering it:
1. init is useful in general for attributes
2. but in bool_attr and friends you may want to use init_val instead
3. however, init can still be useful in bool_attr if the value isn't constant